protoc-gen-grpc-gateway/descriptor/registry.go

```diff
@@ -205,6 +210,11 @@ func (r *Registry) LookupFile(name string) (*File, error) {
 	return f, nil
 }
 
+// LookupExternalHTTPRules looks up external http rules by fully qualified service method name
+func (r *Registry) LookupExternalHTTPRules(qualifiedMethodName string) []*annotations.HttpRule {
```

Comment:
You have a getter for external http rules, I think you should add a setter and use it instead of direct manipulation.